tests/lib/utils/vue-component.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @fileoverview Enforces that a return statement is present in computed property (return-in-computed-property)
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../../../lib/utils/index')
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = {
+  create (context) {
+    return utils.executeOnVueComponent(context, obj => {
+      context.report({
+        node: obj,
+        message: 'Internal test: component.'
+      })
+    })
+  },
+  meta: {
+    fixable: null,
+    schema: []
+  }
+}
+
+const RuleTester = require('eslint').RuleTester
+const parserOptions = {
+  ecmaVersion: 6,
+  sourceType: 'module'
+}
+
+function makeError (line) {
+  return {
+    message: 'Internal test: component.',
+    line,
+    type: 'ObjectExpression'
+  }
+}
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester()
+ruleTester.run('return-in-computed-property', rule, {
+
+  valid: [
+    {
+      filename: 'test.js',
+      code: `export default { }`,
+      parserOptions
+    },
+    {
+      filename: 'test.js',
+      code: `const foo = {}`,
+      parserOptions
+    },
+    {
+      filename: 'test.js',
+      code: `export const foo = {}`,
+      parserOptions
+    },
+    {
+      filename: 'test.js',
+      code: `var foo = {}`,
+      parserOptions
+    },
+    {
+      filename: 'test.js',
```

Comment:
Add some cases with `.jsx` and `.vue` files too that are not to be treated as components by default.